ibtk/src/lagrangian/FEProjector.cpp

```diff
@@ -240,12 +266,14 @@ FEProjector::buildDiagonalL2MassMatrix(const std::string& system_name)
 
         // Extract the FE system and DOF map, and setup the FE object.
         System& system = d_fe_data->getEquationSystems()->get_system(system_name);
-        const int sys_num = system.number();
         DofMap& dof_map = system.get_dof_map();
         FEData::SystemDofMapCache& dof_map_cache = *d_fe_data->getDofMapCache(system_name);
         dof_map.compute_sparsity(mesh);
         FEType fe_type = dof_map.variable_type(0);
-        std::unique_ptr<QBase> qrule = fe_type.default_quadrature_rule(dim);
+        const Order fe_order = fe_type.order;
```

Comment:
Note that this will only work for certain element types.

Perhaps we could use `QNODAL` if available and otherwise throw an error if an element type is not supported.

Also, although I think that `QTRAP` is equivalent to the originally implemented lumping technique for `FIRST` order elements, I don't know whether `QSIMPSON` is equivalent for `SECOND` order elements.

Reply:
`QNODAL` actually turns into `QSIMPSON`: see

https://github.com/libMesh/libmesh/blob/7f11f2f7247d487176cde060fa8c60be35acf402/src/quadrature/quadrature_nodal_2D.C#L138

```cpp
    case QUAD9:
    case TRI6:
      {
        QSimpson rule(/*dim=*/2, /*ignored*/_order);
        rule.init(_type, /*ignored*/_p_level);
        _points.swap (rule.get_points());
        _weights.swap(rule.get_weights());
        return;
      }
```

The correct answer for tensor-product elements is Simpson's rule since, in that case, it is equal to the Gauss-Lobatto rule. It doesn't look like libMesh's triangular Simpson's rule achieves a higher order than composite trapezoid. I'll see if I can find an alternative quadrature rule that can integrate quadratics correctly.

Reply:
We do get different lumped mass matrices when we use `QNODAL` or equivalent but this doesn't matter for velocity interpolation, since in that case the value on the main diagonal of the lumped mass matrix is the same thing we multiply the interpolated velocity by (i.e., if we are doing velocity interpolation, we really do end up interpolating the velocity).

For force spreading this choice does matter: we get a different `JxW` value. Its not clear to me if one should be better than the other. I am a little suspicious of the current implementation because I do not know what quadrature rule it corresponds to, and for a given set of points there should only be one set of weights.